lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -629,159 +629,177 @@ This is problematic for linalg ops for a few reasons:
 h! Although it is a bit like using a hammer to paint, our workaround is to use
 tensor.extract to access the elements of the input tensor inside our linalg
 generic op's payload.
-
-Current TODO's:
-    1. gather most of the boilerplate out of this op and make it into an
-adaptive pooling helper function.
-    2. figure out what to do with the conflicting decompositions in
-DecomposeComplexOps.cpp
-    3. Implement more efficient passes for when the kernel-size, input spatial
-dims, and output spatial dims are constant.
 */
 
 namespace {
-class ConvertAtenAdaptiveAvgPool1dOp
-    : public OpConversionPattern<AtenAdaptiveAvgPool1dOp> {
+
+class AdaptivePoolingHelper {
```

Comment:
"Helper" usually refers to helper functions. I think a better name for this would be something that makes it clear that this eventually templates / inherits down to something that provides an OpConversionPattern.

Reply:
I think it is just used to store some helper functions and help reduce the number of args passed between helper functions by storing them in the class. 

The additional AdaptivePoolingOpsTraits classes are used to get the dim from the optype and alias the helper class name to something generic so it can be referenced in the conversion pattern.